### DISCUSSION TITLE
China

### DISCUSSION THREAD

**Discussion by driveitlikearocket** [IceWhaleTech/CasaOS#1087] [General]
Hey, noticed you guys were based in Shanghai, PRC. Before I use this on my data, what's the CCP involvement in your company? I know they don't generally allow companies to exist that don't conform to them. Is this code base getting certified so we don't have to worry about Chinese back doors? 

It looks awesome. I'd love to use and trust it, but currently even though its open source, it seems it opens some random ports (#512) and that the current state of port forwarding on the firewall is not easily viewable. Going through the requests and issues here in Git, Security seems to come second or third. Updates often aren't working in the UI, 2FA doesn't exist even as an option and is often requested, and there is no brute force protection. Until a certified clean code version is available, the pretty UI must come second to having a real security like UNRAID offers.

**Comment by tigerinus** [comment]
To your first question: No and no.

For the rest of your concerns, this is an open source project heavily relies on community support and involvement after all. We are always looking for people with expertise in security, performance and reliability to contribute to this project. If there is anything you think can be improved, a PR is always welcome. 🙂
